android/engine/src/main/java/org/smartregister/fhircore/engine/domain/model/ResourceData.kt

```diff
@@ -32,6 +39,8 @@ import org.hl7.fhir.r4.model.ResourceType
 data class ResourceData(
   val baseResourceId: String,
   val baseResourceType: ResourceType,
-  val computedValuesMap: Map<String, Any>,
-  val listResourceDataMap: Map<String, List<ResourceData>>? = null
+  val computedValuesMap: SnapshotStateMap<String, Any>,
+  val listResourceDataMap: SnapshotStateMap<String, SnapshotStateList<ResourceData>> =
+    mutableStateMapOf(),
+  val baseResource: Resource? = null
```

Comment:
ResourceData  here will hold reference to the base resource which is not used anywhere in the rendered view thus leaking memory at this point.

Reply:
Probably, :grimacing: we are probably using more memory but not leaking it. We need also need to the baseResource for the profile rendering and recomputations